numpy/lib/function_base.py

```diff
@@ -1173,6 +1179,29 @@ def diff(a, n=1, axis=-1):
             "order must be non-negative but got " + repr(n))
 
     a = asanyarray(a)
+
+    combined = []
+    if prepend is not np._NoValue:
+        prepend = np.asanyarray(prepend)
+        if prepend.ndim == 0:
+            shape = list(a.shape)
+            shape[axis] = 1
+            prepend = np.broadcast_to(prepend, tuple(shape))
+        combined.append(prepend)
+
+    combined.append(a)
+
+    if append is not np._NoValue:
+        append = np.asanyarray(append)
+        if append.ndim == 0:
+            shape = list(a.shape)
+            shape[axis] = 1
```

Comment:
... otherwise this will throw an IndexError, rather than the more useful `AxisError`